### PR TITLE
Implement Notebook and Notification export on the client side like the other exports

### LINKED ISSUE #2049: Implement Notebook and Notification export on the client side like the other exports
### Description

Right now the Notebook and Notification exports are generated on the server. The other exports are generated on the client. For the other exports, we send all the data to the client and then have the client process it and generate a CSV file. We want to make the Notebook and Notification exports generated on the client as well.

After this is done, these server functions can be removed
```
HibernateNotificationDao.java getNotificationExport()
HibernateNotebookItemDao.java getNotebookItemExport()
```

### PR DESCRIPTION
**Pull request by geoffreykwan**
Make sure the Notifications export works properly.

Make sure the Latest and All Notebook Items export works properly.

Closes #2049

### CODE REVIEW

**Comment by hirokiterashima** [comment]
@geoffreykwan should this be merged into the issue-851-update-hibernate-criteria branch? 

If/when this get merged into develop first, the work in the hibernate-criteria branch might be swept into develop without thorough testing.

File: src/main/java/org/wise/portal/dao/notification/impl/HibernateNotificationDao.java [hunk 1]
```diff
@@ -132,4 +102,18 @@ protected String getFindAllQuery() {
...
+  public List<Notification> getExport(Run run) {
+    CriteriaBuilder cb = getCriteriaBuilder();
+    CriteriaQuery<Notification> cq = cb.createQuery(Notification.class);
+    Root<Notification> notificationRoot = cq.from(Notification.class);
+    List<Predicate> predicates = new ArrayList<>();
+    predicates.add(cb.equal(notificationRoot.get("run"), run));
+    cq.select(notificationRoot).where(predicates.toArray(new Predicate[predicates.size()]))
+        .orderBy(cb.asc(notificationRoot.get("toWorkgroup")), cb.asc(notificationRoot.get("id")));
+    TypedQuery<Notification> query = entityManager.createQuery(cq);
+    return (List<Notification>)(Object)query.getResultList();
```
Comment (hirokiterashima) [inline src/main/java/org/wise/portal/dao/notification/impl/HibernateNotificationDao.java]:
```suggestion
    return (List<Notification>) (Object) query.getResultList();
```

File: src/main/java/org/wise/portal/dao/work/impl/HibernateNotebookItemDao.java [hunk 2]
```diff
@@ -100,30 +99,40 @@ protected String getFindAllQuery() {
...
+  @SuppressWarnings("unchecked")
+  public List<NotebookItem> getNotebookItemsExport(Run run) {
+    CriteriaBuilder cb = getCriteriaBuilder();
+    CriteriaQuery<NotebookItem> cq = cb.createQuery(NotebookItem.class);
+    Root<NotebookItem> notebookItemRoot = cq.from(NotebookItem.class);
+    cq.select(notebookItemRoot).where(cb.equal(notebookItemRoot.get("run"), run))
+        .orderBy(cb.asc(notebookItemRoot.get("workgroup").get("id")),
+        cb.asc(notebookItemRoot.get("id")));
+    TypedQuery<NotebookItem> query = entityManager.createQuery(cq);
+    return (List<NotebookItem>)(Object)query.getResultList();
```
Comment (hirokiterashima) [inline src/main/java/org/wise/portal/dao/work/impl/HibernateNotebookItemDao.java]:
```suggestion
    return (List<NotebookItem>) (Object) query.getResultList();
```

File: src/main/java/org/wise/portal/dao/work/impl/HibernateNotebookItemDao.java [hunk 3]
```diff
@@ -100,30 +99,40 @@ protected String getFindAllQuery() {
...
+    Root<NotebookItem> notebookItemRoot = cq.from(NotebookItem.class);
+    Subquery<Long> latestNotebookItemIds = getLatestNotebookItemIds(cb, cq);
+    List<Predicate> predicates = new ArrayList<>();
+    predicates.add(cb.equal(notebookItemRoot.get("run"), run));
+    predicates.add(notebookItemRoot.get("id").in(latestNotebookItemIds));
+    cq.select(notebookItemRoot).where(predicates.toArray(new Predicate[predicates.size()]))
+        .orderBy(cb.asc(notebookItemRoot.get("workgroup").get("id")),
+        cb.asc(notebookItemRoot.get("id")));
+    TypedQuery<NotebookItem> query = entityManager.createQuery(cq);
+    return (List<NotebookItem>)(Object)query.getResultList();
```
Comment (hirokiterashima) [inline src/main/java/org/wise/portal/dao/work/impl/HibernateNotebookItemDao.java]:
```suggestion
    return (List<NotebookItem>) (Object) query.getResultList();
```

File: src/main/java/org/wise/portal/service/vle/wise5/impl/VLEServiceImpl.java [hunk 4]
```diff
@@ -138,54 +137,46 @@
...
+  public JSONArray getNotebookItemsExport(Integer runId) {
+    try {
+      Run run = runService.retrieveById(new Long(runId));
+      List<NotebookItem> notebookItems = notebookItemDao.getNotebookItemsExport(run);
+      JSONArray notebookItemsJSONArray = new JSONArray();
+      for (int n = 0; n < notebookItems.size(); n++) {
+        notebookItemsJSONArray.put(notebookItems.get(n).toJSON());
       }
+      return notebookItemsJSONArray;
+    } catch(Exception e) {
```
Comment (hirokiterashima) [inline src/main/java/org/wise/portal/service/vle/wise5/impl/VLEServiceImpl.java]:
```suggestion
    } catch (Exception e) {
```

File: src/main/java/org/wise/portal/service/vle/wise5/impl/VLEServiceImpl.java [hunk 5]
```diff
@@ -138,54 +137,46 @@
...
+  public JSONArray getLatestNotebookItemsExport(Integer runId) {
+    try {
+      Run run = runService.retrieveById(new Long(runId));
+      List<NotebookItem> notebookItems = notebookItemDao.getLatestNotebookItemsExport(run);
+      JSONArray notebookItemsJSONArray = new JSONArray();
+      for (int n = 0; n < notebookItems.size(); n++) {
+        notebookItemsJSONArray.put(notebookItems.get(n).toJSON());
       }
+      return notebookItemsJSONArray;
+    } catch(Exception e) {
```
Comment (hirokiterashima) [inline src/main/java/org/wise/portal/service/vle/wise5/impl/VLEServiceImpl.java]:
```suggestion
    } catch (Exception e) {
```

**Comment by geoffreykwan** [comment]
Ok, I've added the spaces when casting.

**Pull request merged**